### PR TITLE
MISC: Show project structure in dump mode #2824

### LINKED ISSUE #2824: Show project structure in dump mode
Currently, we don't show project structure while indexing
screen shot 2018-09-05 at 11 51 10

It can be inconvenient for users especially when a project is quite big and indexing takes a lot of time.
Looks like we can show project structure in dumb mode because we get project metadata from cargo directly which doesn't depend on IDE indexing

### PR DESCRIPTION
**Pull request by amakeev**
also make cargo/project-related actions DumbAware

Closes #2824

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/project/model/CargoProjectActionBase.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.model
+
+import com.intellij.openapi.project.DumbAwareAction
+
+abstract class CargoProjectActionBase : DumbAwareAction() {
+}
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/project/model/CargoProjectActionBase.kt]: I think we can omit `{}` here

File: src/main/kotlin/org/rust/cargo/project/model/DetachCargoProjectAction.kt [hunk 2]
```diff
@@ -9,7 +9,7 @@ import com.intellij.openapi.actionSystem.AnAction
 import com.intellij.openapi.actionSystem.AnActionEvent
 import org.rust.cargo.project.toolwindow.CargoToolWindow
 
-class DetachCargoProjectAction : AnAction() {
+class DetachCargoProjectAction : CargoProjectActionBase() {
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/project/model/DetachCargoProjectAction.kt]: Unused import of `com.intellij.openapi.actionSystem.AnAction`

File: src/main/kotlin/org/rust/cargo/runconfig/command/RunCargoCommandActionBase.kt [hunk 3]
```diff
@@ -8,13 +8,14 @@ package org.rust.cargo.runconfig.command
...
 import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.project.DumbAwareAction
 import org.rust.cargo.project.model.CargoProject
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.cargo.project.toolwindow.CargoToolWindow
 import org.rust.cargo.runconfig.hasCargoProject
 import javax.swing.Icon
 
-abstract class RunCargoCommandActionBase(icon: Icon) : AnAction(icon) {
+abstract class RunCargoCommandActionBase(icon: Icon) : DumbAwareAction(null, null, icon) {
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/runconfig/command/RunCargoCommandActionBase.kt]: Unused import of `com.intellij.openapi.actionSystem.AnAction`

File: src/main/kotlin/org/rust/ide/actions/RefreshCargoProjectAction.kt [hunk 4]
```diff
@@ -7,13 +7,14 @@ package org.rust.ide.actions
...
 import com.intellij.openapi.actionSystem.AnActionEvent
+import org.rust.cargo.project.model.CargoProjectActionBase
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.cargo.project.model.guessAndSetupRustProject
 import org.rust.cargo.project.settings.toolchain
 import org.rust.cargo.runconfig.hasCargoProject
 import org.rust.openapiext.saveAllDocuments
 
-class RefreshCargoProjectAction : AnAction() {
+class RefreshCargoProjectAction : CargoProjectActionBase() {
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/actions/RefreshCargoProjectAction.kt]: Unused import of `com.intellij.openapi.actionSystem.AnAction`

**Comment by Undin** [comment]
@amakeev just write `bors r+` to merge PR after fixing comments

File: src/main/kotlin/org/rust/cargo/project/model/CargoProjectActionBase.kt [hunk 5]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.project.model
+
+import com.intellij.openapi.project.DumbAwareAction
+
+abstract class CargoProjectActionBase : DumbAwareAction()
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/project/model/CargoProjectActionBase.kt]: :thinking: Missed linebreak at the end of the file

Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/project/model/CargoProjectActionBase.kt]: Still no linebreak

**Comment by vlad20012** [comment]
Also looks like you accidentally added commit with CONTRIBUTORS.txt
image

**Comment by amakeev** [comment]
@vlad20012 Indeed, thanks, was incorrectly added obviously.

**Comment by vlad20012** [comment]
Could you please squash these commits before r+? :)

**Pull request merged**